ddl/generated_column.go

```diff
@@ -41,21 +41,24 @@ func verifyColumnGeneration(colName2Generation map[string]columnGenerationInDDL,
 					return errors.Trace(err)
 				}
 			} else {
-				err := errBadField.GenWithStackByArgs(depCol, "generated column function")
+				err := ErrBadField.GenWithStackByArgs(depCol, "generated column function")
 				return errors.Trace(err)
 			}
 		}
 	}
 	return nil
 }
 
-// columnNamesCover checks whether dependColNames is covered by normalColNames or not.
-// it's only for alter table add column because before alter, we can make sure that all
-// columns in table are verified already.
-func columnNamesCover(normalColNames map[string]struct{}, dependColNames map[string]struct{}) error {
-	for name := range dependColNames {
-		if _, ok := normalColNames[name]; !ok {
-			return errBadField.GenWithStackByArgs(name, "generated column function")
+// ensure all depended columns are exist.
+//
+// NOTE: this will destroy parameter `dependCols`
```

Comment:
IMHO it's better to replace term 'destroy' with 'modify'... 'destroy' a map is confusing